hack/generator/pkg/codegen/code_generator.go

```diff
@@ -104,8 +115,94 @@ func (generator *CodeGenerator) Generate(ctx context.Context, outputFolder strin
 	return nil
 }
 
-func (generator *CodeGenerator) CreatePackagesForDefinitions(definitions []astmodel.TypeDefiner) ([]*astmodel.PackageDefinition, error) {
-	packages := make(map[astmodel.PackageReference]*astmodel.PackageDefinition)
+func (generator *CodeGenerator) MarkLatestResourceVersionsForStorage(pkgs []*astmodel.PackageDefinition) ([]*astmodel.PackageDefinition, error) {
+
+	var result []*astmodel.PackageDefinition
+
+	resourceLookup, err := groupResourcesByVersion(pkgs)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, pkg := range pkgs {
+
+		resultPkg := astmodel.NewPackageDefinition(pkg.GroupName, pkg.PackageName)
+		for _, def := range pkg.Definitions() {
+			if structDef, ok := def.(*astmodel.StructDefinition); ok && structDef.IsResource() {
+
+				unversionedName, err := getUnversionedName(structDef.TypeName)
+				if err != nil {
+					// should never happen as all resources have versioned names
+					return nil, err
+				}
+
+				allVersionsOfResource := resourceLookup[unversionedName]
+				latestVersionOfResource := allVersionsOfResource[len(allVersionsOfResource)-1]
+
+				isLatestVersion := structDef.Name().PackageReference.PackagePath() == latestVersionOfResource.Name().PackageReference.PackagePath()
+				structDef = structDef.WithIsStorageVersion(isLatestVersion)
+
+				resultPkg.AddDefinition(structDef)
+			} else {
+				resultPkg.AddDefinition(def)
+			}
+		}
+
+		result = append(result, resultPkg)
+	}
+
+	return result, nil
+}
+
+func getUnversionedName(name *astmodel.TypeName) (unversionedName, error) {
+
```

Comment:
super minor: Some methods you've left a newline at the start of, and others you haven't, and I can't tell why =)

I personally don't love the newline, but mostly just wondering why the inconsistency?

Reply:
no reason; sometimes it seemed cleaner. should be consistent now; have left a blank line after a wrapped function declaration, otherwise none